File: app/src/main/java/com/viafourasample/src/activities/login/LoginViewModel.java
package com.viafourasample.src.activities.login;

import com.onesignal.OneSignal;
import com.viafourasdk.src.ViafouraSDK;
import com.viafourasdk.src.model.network.authentication.login.LoginResponse;
import com.viafourasdk.src.model.network.authentication.socialLogin.SocialLoginResponse;
import com.viafourasdk.src.model.network.error.NetworkError;
import com.viafourasdk.src.services.auth.VFAuthService;

public class LoginViewModel {
    private VFAuthService auth = ViafouraSDK.auth();

    public void login(String email, String password, LoginCallback callback){
        auth.login(email, password, new VFAuthService.LoginCallback() {
            @Override
            public void onSuccess(LoginResponse loginResponse) {
                OneSignal.login(String.valueOf(loginResponse.result.id));
                callback.onSuccess();
            }

            @Override
            public void onError(NetworkError err) {
                callback.onError(err.message);
            }
        });
    }

    interface LoginCallback {
        void onSuccess();
        void onError(String errorMessage);
    }

    public void resetPassword(String email, PasswordResetCallback callback){
        auth.passwordReset(email, new VFAuthService.PasswordResetCallback() {
            @Override
            public void onSuccess() {
                callback.onSuccess();
            }

            @Override
            public void onError(NetworkError err) {
                callback.onError();
            }
        });
    }

    interface PasswordResetCallback {
        void onSuccess();
        void onError();
    }

    public void socialLogin(String token, SocialLoginCallback socialLoginCallback){
        auth.socialLogin(token, new VFAuthService.SocialLoginCallback() {
            @Override
            public void onSuccess(SocialLoginResponse loginResponse) {
                socialLoginCallback.onSuccess();
            }

            @Override
            public void onError(NetworkError err) {
                socialLoginCallback.onError();
            }
        });
    }

    interface SocialLoginCallback {
        void onSuccess();
        void onError();
    }
}

File: app/src/main/java/com/viafourasample/src/activities/login/LoginViewModel.kt
package com.viafourasample.src.activities.login

import com.viafourasdk.src.ViafouraSDK
import com.viafourasdk.src.model.network.authentication.login.LoginResponse
import com.viafourasdk.src.model.network.error.NetworkError
import com.viafourasdk.src.services.auth.VFAuthService

class LoginViewModel {
    private val auth: VFAuthService = ViafouraSDK.auth()

    fun login(email: String, password: String, callback: LoginCallback) {
        auth.login(email, password, object : VFAuthService.LoginCallback {
            override fun onSuccess(loginResponse: LoginResponse) {
                callback.onSuccess()
            }

            override fun onError(err: NetworkError) {
                callback.onError(err.message)
            }
        })
    }

    fun resetPassword(email: String, callback: PasswordResetCallback) {
        auth.passwordReset(email, object : VFAuthService.PasswordResetCallback {
            override fun onSuccess() {
                callback.onSuccess()
            }

            override fun onError(err: NetworkError) {
                callback.onError()
            }
        })
    }

    interface LoginCallback {
        fun onSuccess()
        fun onError(errorMessage: String?)
    }

    interface PasswordResetCallback {
        fun onSuccess()
        fun onError()
    }
}
